lib/transformer/graph_methods.rb

```diff
@@ -0,0 +1,42 @@
+module Transformer
+  module GraphMethods
+    def self.all
+      @graph_methods ||= begin
+        result = {}
+
+        Atlas::Node.all.each do |node|
+          node.graph_methods.each do |method|
+            result[:"#{ node.key }_#{ method }"] = {
+              type: Float,
+              export: {
+                key: node.key,
+                method: method
+              }
+            }
+          end
+        end
+
+        Atlas::Edge.all.each do |edge|
+          edge.graph_methods.each do |method|
+            key = case method
+                  when 'child_share'
+                    edge.consumer
+                  when 'parent_share'
+                    edge.supplier
+                  end
+
+            result[:"#{ key }_#{ method }"] = {
```

Comment:
Is it correct to use the consumer/supplier key here? What happens if there are several edges?

For example:

* x → y (parent_share)
* x → z (parent_share)

Does this not result in a single `x_parent_share` value being added to `result`?

Reply:
The idea is to have:

```
z_parent_share
y_parent_share
```

To make the keys unique. I appearantly swapped them around yesterday since I made them `x_parent_share` ... Oops.

Reply:
> The idea is to have: `z_parent_share`, `y_parent_share` [...] To make the keys unique.

Correct me if I'm wrong, but I'm not sure that the keys would be unique.

![untitled diagram 2](https://user-images.githubusercontent.com/4383/33880824-c70327a2-df2a-11e7-9815-6f327d60498c.png)

Would the red and blue edges not have conflicting keys?

Reply:
> Would the red and blue edges not have conflicting keys?

Fair point, yes they would 👍